plenum/test/txn_author_agreement/test_config_req_handler_taa_utils.py

```diff
@@ -31,61 +37,107 @@ def test_is_trustee(txnPoolNodeSet, sdk_wallet_trustee, sdk_wallet_steward, sdk_
+            if digest is not None:
+                digest = digest.encode()
+            assert digest == state.get(
+                ConfigReqHandler._state_path_taa_version(version), isCommitted=False)
 
 
-def test_multiple_update_txn_author_agreement(config_req_handler: ConfigReqHandler):
-    text_v1 = 'Some agreement'
-    agreement_v1 = {TXN_AUTHOR_AGREEMENT_VERSION: 'v1', TXN_AUTHOR_AGREEMENT_TEXT: text_v1}
-    digest_v1 = ConfigReqHandler._taa_digest('v1', text_v1).encode()
-    config_req_handler.update_txn_author_agreement(agreement_v1)
-
-    assert config_req_handler.get_taa_digest(isCommitted=False) == digest_v1
-    assert config_req_handler.get_taa_digest(version='v1', isCommitted=False) == digest_v1
-    assert config_req_handler.get_taa_digest(version='v2', isCommitted=False) is None
-
-    text_v2 = 'New agreement'
-    agreement_v2 = {TXN_AUTHOR_AGREEMENT_VERSION: 'v2', TXN_AUTHOR_AGREEMENT_TEXT: text_v2}
-    digest_v2 = ConfigReqHandler._taa_digest('v2', text_v2).encode()
-    config_req_handler.update_txn_author_agreement(agreement_v2)
+def test_get_taa_data(config_req_handler: ConfigReqHandler,
+                      taa_in_data, taa_state_data, taa_digests):
+    """ `get_taa_data` returns values from state """
+    state = config_req_handler.state
 
-    assert config_req_handler.get_taa_digest(isCommitted=False) == digest_v2
-    assert config_req_handler.get_taa_digest(version='v1', isCommitted=False) == digest_v1
-    assert config_req_handler.get_taa_digest(version='v2', isCommitted=False) == digest_v2
+    def _state_data(digest=None, version=None):
+        if digest is None:
+            digest = config_req_handler.get_taa_digest(
+                version=version, isCommitted=False)
+        data = state.get(
+            ConfigReqHandler._state_path_taa_digest(digest),
+            isCommitted=False
+        )
+        return None if data is None else config_state_serializer.deserialize(data)
+
+    for data in taa_in_data:
+        config_req_handler.update_txn_author_agreement(*data)
+
+        assert (
+            config_req_handler.get_taa_data(isCommitted=False) == _state_data()
+        )
+
+        for version in taa_state_data:
+            data = config_req_handler.get_taa_data(version=version, isCommitted=False)
+            assert data == _state_data(version=version)
+
+        for digest in taa_digests:
+            data = config_req_handler.get_taa_data(
+                digest=digest, version='any-version-since-ignored',
+                isCommitted=False
+            )
+            assert data == _state_data(digest=digest)
+
+
+def test_multiple_update_txn_author_agreement(
+        config_req_handler: ConfigReqHandler, taa_in_data, taa_digests, taa_state_data):
+    written = []
+    for data in taa_in_data:
+        digest = taa_digests[data.version]
+
+        config_req_handler.update_txn_author_agreement(*data)
+        written.append(data.version)
```

Comment:
Feels like this test contains too much logic compared to original one with just two examples.

Reply:
updated